### PR TITLE
Boolean model

### PR DESCRIPTION
**Pull request by JihaneBennis**
boolean method 
- Handles "AND" first
- Handles "NOT" and "OR" elements
feel free to test :)

### CODE REVIEW

File: tokenisation.py [hunk 1]
```diff
@@ -1,9 +1,79 @@
 import os
 import nltk
+import pdb
 from text_processing import create_lowercase_text, tokenize, vocabulary
 from utils import create_inverted_index, compute_linear_reg, plot_frequecy_distribution
 
-path = os.getcwd()+'\\Data\\CACM\\cacm.all'
+# path = os.getcwd()+'\\Data\\CACM\\cacm.all'
+path = "./Data/CACM/cacm.all"a
```
Comment (selimbat) [inline tokenisation.py]: `path = "./Data/CACM/cacm.all"`

Comment (selimbat) [reply tokenisation.py]: sans le a à la fin

File: tokenisation.py [hunk 2]
```diff
@@ -1,9 +1,79 @@
 import os
```
Comment (selimbat) [inline tokenisation.py:1]: import re

File: tokenisation.py [hunk 3]
```diff
@@ -38,6 +132,116 @@
...
+    request = tokenize(client_request)
+    final_docs=[]
+    documents = []
+    #forbidden_doc = []
+    #handling AND elements first
+    while(next((operator for operator in request if operator == 'AND'), False)):
+        index_treated = []
+        forbidden_doc = list(set())
+        i = request.index('AND')
+        print('AND loop, i:', i, ' request : ', request)
```
Comment (selimbat) [inline tokenisation.py]: delete print

File: tokenisation.py [hunk 4]
```diff
@@ -38,6 +132,116 @@
...
+            if type(request[i+1])==str:
+                doc_treated= inverted_index[request[i + 1]]
+            else:
+                doc_treated = request[i+1]
+            if(documents != []):
+                documents=intersection(documents,doc_treated)
+            else:
+                documents+=doc_treated
+        if len(forbidden_doc) > 0: # Removing forbidden docs from results
+            print('forbidden docs',forbidden_doc)
```
Comment (selimbat) [inline tokenisation.py]: delete print

File: tokenisation.py [hunk 5]
```diff
@@ -38,6 +132,116 @@
...
+            request.pop(index_treated[1])
+        final_docs=documents
+    return list(set(final_docs))
+
+'''example of request'''
+
+print('mechanical : ',inverted_index['mechanical'])
+print('pragmatics :', inverted_index['pragmatics'])
+print('police : ', inverted_index['police'])
+request_andNot = 'pragmatics AND NOT pragmatics AND police'
```
Comment (selimbat) [inline tokenisation.py:236]: This peticular request for instance returns [] instead of inverted_index['police'] as it should theoretically

Comment (JihaneBennis) [reply tokenisation.py:236]: resolved :)

**Pull request merged**